app/api/validations.py

```diff
@@ -24,9 +24,19 @@ def missing_json_error():
     return standardize_response(error, status_code=422)
 
 
-def validate_resource(request, id=-1):
+def validate_resource_list(req, rlist):
```

Comment:
Another case that both list and dict makes it feel awkward. 

Above we call `validate_resource` and then we call `validate_resource_list` but in both cases we call the returned value `resource_errors` 

If we have a list of resources does this mean we could have a list of errors for each resource (or an empty list) and a list of errors for a single resource?

Reply:
Well the beauty of the errors is that each resource will have an error object, which is how the API responds right now for errors. When you try to create a single resource and it fails validation, it gives you an object with all of the errors inside. Now, when you try to create a batch of resources and several of them have errors, you get an error object for each invalid resource. The problem is mapping the error object to the submitted error because the user doesn't submit an identifier. You could use the URL as a unique identifier, except that one way that validation fails is that you may not have a URL at all... so I just did it by index. I really couldn't figure out a better way that evening, and I haven't thought of anything better since 🤷‍♂

Reply:
I also wonder what happens if the errors are different for each type. Or if you would want to create the resources that are okay and only return error resources. But that would create a mixed-mode response.

Reply:
I refuse to let the API create some resources but not all. I prefer full idempotency, but that's a tall order. But I think you should be able to fix the problems with your json and then send the same request with those issues fixed